### PR TITLE
Modular helmet binocs

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

title
## Why It's Good For The Game

Helmet time
## Changelog
:cl:
add: Added a modular helmet binoculars module that is not yet available in vendors
/:cl:

### CODE REVIEW

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -667,7 +667,7 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 		to_chat(user, "<span class='warning'>Your vision is too obscured for you to look through \the [zoom_device].</span>")
 		return
 
-	if(!zoom && user.get_active_held_item() != src)
+	if(!zoom && user.get_active_held_item() != src && !istype(src, /obj/item/helmet_module/binoculars))
```
Comment (Rohesie) [inline code/game/objects/items.dm:670]: Add this as a flag instead. Then do something like this:
```suggestion
	if(!zoom && flags_item & REQUIRES_HANDS_TO_USE && user.get_active_held_item() != src)
```

File: code/modules/modular_armor/helmets.dm [hunk 2]
```diff
@@ -58,3 +58,23 @@
...
+	name = "Binocular Helmet Module"
+	icon_state = "binocular_head_obj"
+	item_state = "binocular_head"
+	module_type = ARMOR_MODULE_TOGGLE
+	active = FALSE
+
+/obj/item/helmet_module/binoculars/toggle_module(mob/living/user, obj/item/clothing/head/modular/parent)
+	if(!active && !zoom)
+		RegisterSignal(user, COMSIG_MOVABLE_MOVED, .proc/toggle_module)
+		RegisterSignal(user, COMSIG_CLIENT_MOUSEDOWN, .proc/toggle_module)//No shooting while zoomed
```
Comment (Rohesie) [inline code/modules/modular_armor/helmets.dm:72]:
```suggestion
		RegisterSignal(user, list(COMSIG_MOVABLE_MOVED, COMSIG_CLIENT_MOUSEDOWN), .proc/toggle_module) //No shooting while zoomed
```

File: code/modules/modular_armor/helmets.dm [hunk 3]
```diff
@@ -58,3 +58,23 @@
...
+		zoom(user, 11, 12)
+	else
+		UnregisterSignal(user, COMSIG_MOVABLE_MOVED)
```
Comment (Rohesie) [inline code/modules/modular_armor/helmets.dm:75]:
```suggestion
		UnregisterSignal(user, list(COMSIG_MOVABLE_MOVED, COMSIG_CLIENT_MOUSEDOWN))
```

**Comment by psykzz** [comment]
still no sprites

**Comment by TiviPlus** [comment]
Sprites were added in a different PR, ill update the description

**Comment by psykzz** [comment]
Ah okay then

**Comment by jroinc** [comment]
Add a delay on use for this one. Otherwise it's just more unga GAMER GEAR that has no downsides that everyone carries. It's just a binoculars that protects your head, can carry 2 more things, AND is hands-free.

1-2 second delay seems reasonable.

**Pull request merged**